IpmiFeaturePkg/Include/Library/IpmiSelLib.h

```diff
@@ -0,0 +1,143 @@
+/** @file
+  Definitions for the SEL Library.
+
+  Copyright (c) Microsoft Corporation
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef _IPMI_SEL_LIB_H
+#define _IPMI_SEL_LIB_H
+
+//
+// Structure used for get info results.
+//
+
+#pragma pack(1)
+
+typedef struct _SEL_INFO {
+  UINT8     Version;
+  UINT16    NumberOfEntries;
+  UINT16    FreeSpace;
+  UINT32    LastAddTimeStamp;
+  UINT32    LastEraseTimeStamp;
+  union {
+    struct {
+      UINT8    GetSelAllocationInfo : 1;
+      UINT8    ReserveSel           : 1;
+      UINT8    PartialAddSelEntry   : 1;
+      UINT8    DeleteSel            : 1;
+      UINT8    Reserved             : 3;
+      UINT8    Overflow             : 1;
+    } Bits;
+
+    UINT8    AsUint8;
+  } OperationSupported;
+} SEL_INFO;
+
+#pragma pack()
+
+/**
+  Adds a system event to the SEL.
+
+  @param[in,out]  RecordId      If provided, receives the record ID of the entry.
+  @param[in]      SensorType    The Sensor type for the event.
+  @param[in]      SensorNumber  The sensor number for the event.
+  @param[in]      EventDirType  The event Dir and Type values
+  @param[in]      Data0         OEM defined data part 0.
+  @param[in]      Data1         OEM defined data part 1.
+  @param[in]      Data2         OEM defined data part 2.
+
+  @retval   EFI_SUCCESS     Event was successfully added to the SEL.
+  @retval   Other           And error was returned by IpmiAddSelEntry.
+**/
+EFI_STATUS
+EFIAPI
+SelAddSystemEntry (

```

Comment:
I believe this is part of the spec, you can either have System entries (code 0x2) or OEM entries

Reply:
I see, I think I meant the record formats in section 32. sel record format. they have the oem sel records (timestamped and no timestamp) and regular sel event records.

Reply:
True, but regular only has one defined type which is system. So this format is refered to as system and since there is no other "regular" type in IPMI v2 and IPMI is now a dead spec I didn't see any point in allowing generic non-spec-compliant types. It doesnt seem its used that way anyways from every thing i've seen.